### DISCUSSION TITLE
Offline exchange on Android

### DISCUSSION THREAD

**Discussion by lsarni** [urql-graphql/urql#3654] [Q&A]
We are using `offlineExchange` with `AsyncStorage` for React Native as explained [here](https://commerce.nearform.com/open-source/urql/docs/graphcache/offline/#react-native).
But we are running into this issue on some Android devices, even after applying the suggested workaround https://github.com/react-native-async-storage/async-storage/issues/1110

I was wondering if anyone had found a workaround for this that allowed them to keep the offline caching.

**Comment by kitten** [comment]
Which of the two limits are you running into? It looks like total storage can be increased, but the per entry limit cannot.

The web default storage has a few strategies for splitting up values that may be applicable here, but I'm not sure what the current state of our RN adapter is.
In all likelihood, this is definitely fixable, but i'd need a better understanding of the problem to be sure of that ✌️
